lib/cache/policy.js

```diff
@@ -97,6 +98,9 @@ class CachePolicy {
     if (this.request.headers.host !== _req.headers.host)
       return false
 
+    if (this.request.compress !== request.compress)
```

Comment:
since you've added the `compress` option as part of the result in the `requestObject()` function above, this should probably compare to `_req.compress` instead of `request.compress` for consistency

Reply:
Yes, indeed. Will fix.